src/trace/implementations/ord.rs

```diff
@@ -220,18 +221,20 @@ where
     R: Semigroup,
     O: OrdOffset, <O as TryFrom<usize>>::Error: Debug, <O as TryInto<usize>>::Error: Debug
 {
-	fn new(batch1: &OrdValBatch<K, V, T, R, O>, batch2: &OrdValBatch<K, V, T, R, O>) -> Self {
+	fn new(batch1: &OrdValBatch<K, V, T, R, O>, batch2: &OrdValBatch<K, V, T, R, O>, compaction_frontier: Option<&[T]>) -> Self {
 
 		assert!(batch1.upper() == batch2.lower());
 
-		let since = if batch1.description().since().iter().all(|t1| batch2.description().since().iter().any(|t2| t2.less_equal(t1))) {
-			batch2.description().since()
-		}
-		else {
-			batch1.description().since()
+		let since = {
+			let since = antichain_join(batch1.description().since(), batch2.description().since());
+			if let Some(compaction_frontier) = compaction_frontier {
+				antichain_join(since.elements(), compaction_frontier)
+			} else {
+				since
+			}
```

Comment:
I wonder if this looks better as
```rust
let mut since = antichain_join(batch1.description().since(), batch2.description().since());
if let Some(compaction_frontier) = compaction_frontier {
   since = antichain_join(since.elements(), compaction_frontier);
}
```
I'm happy to clean that up in post, as long as you don't mind!

Reply:
yeah it definitely does, feel free to refactor